lib/view/widgets/extended_menu_fab.dart

```diff
@@ -0,0 +1,128 @@
+import 'package:flutter/material.dart';
+import 'package:superformula_test/core/resources/app_routes.dart';
+
+class ExtendedMenuFAB extends StatefulWidget {
+  const ExtendedMenuFAB({super.key});
+
+  @override
+  State<ExtendedMenuFAB> createState() => _ExtendedMenuFABState();
+}
+
+class _ExtendedMenuFABState extends State<ExtendedMenuFAB>
+    with SingleTickerProviderStateMixin {
+  late final AnimationController controller;
+  late final Animation<double> animation;
+  bool openedMenu = false;
+
+  @override
+  void initState() {
+    super.initState();
+    controller = AnimationController(
+      vsync: this,
+      duration: const Duration(milliseconds: 500),
+    );
+
+    animation = Tween<double>(begin: 0, end: 150).animate(controller);
+  }
+
+  @override
+  void dispose() {
+    controller.dispose();
+    super.dispose();
+  }
+
+  void onPressFAB() {
+    if (animation.value == 0) {
+      controller.forward();
+    } else {
+      controller.reverse();
+    }
+
+    setState(() {
+      openedMenu = !openedMenu;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Stack(
+      alignment: Alignment.center,
+      clipBehavior: Clip.none,
+      children: [
+        Positioned(
+          bottom: 4,
+          right: 44,
+          child: AnimatedBuilder(
+            animation: animation,
+            child: Row(
+              crossAxisAlignment: CrossAxisAlignment.center,
+              mainAxisAlignment: MainAxisAlignment.center,
+              mainAxisSize: MainAxisSize.min,
+              children: [
+                const Flexible(
+                  child: VerticalDivider(
+                    thickness: 2,
+                    indent: 4,
+                  ),
+                ),
+                Expanded(
+                  flex: 2,
+                  child: _ExtendedMenuItemFAB(
+                    icon: Icons.qr_code_scanner_outlined,
+                    onPressed: () =>
+                        Navigator.of(context).pushNamed(AppRoutesName.scan),
+                  ),
+                ),
+                Expanded(
+                  flex: 2,
+                  child: _ExtendedMenuItemFAB(
+                    icon: Icons.qr_code_2_outlined,
+                    onPressed: () =>
+                        Navigator.of(context).pushNamed(AppRoutesName.qrcode),
+                  ),
+                ),
+              ],
+            ),
+            builder: (context, child) => SizedBox(
```

Comment:
**suggestion**: a more standard approach for this use-case is to use the `SizeTransiton` widget 🙂

Reply:
Agreed! I did not think about that.